### PR TITLE
Embed some linker options to static frameworks

### PR DESCRIPTION
**Pull request by SvyatoslavScherbina**

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CompilerOutput.kt [hunk 1]
```diff
@@ -121,3 +127,24 @@ private fun parseAndLinkBitcodeFile(llvmModule: LLVMModuleRef, path: String) {
...
 }
+
+private fun embedAppleLinkerOptionsToBitcode(llvm: Llvm, config: KonanConfig) {
+    fun findEmbeddableOptions(options: List<String>): List<List<String>> {
+        val result = mutableListOf<List<String>>()
+        val iterator = options.iterator()
+        loop@while (iterator.hasNext()) {
+            val option = iterator.next()
+            result += when {
+                option.startsWith("-l") -> listOf(option)
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CompilerOutput.kt]: What about `-l foo`?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/LlvmLinkOptions.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.jetbrains.kotlin.backend.konan.llvm
+
+import llvm.LLVMAddNamedMetadataOperand
+import llvm.LLVMModuleRef
+
+fun embedLlvmLinkOptions(module: LLVMModuleRef, options: List<List<String>>) {
+    options.forEach {
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/LlvmLinkOptions.kt]: Shalln't we assert that it only works on Apple targets?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CompilerOutput.kt (hunk 1 above)
Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CompilerOutput.kt]: Not supported by the linker.

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/LlvmLinkOptions.kt (hunk 2 above)
Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/LlvmLinkOptions.kt]: This is kinda not true.

Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/LlvmLinkOptions.kt]: You mean, we could use it on other platforms? Like to fix https://github.com/JetBrains/kotlin-native/issues/2803

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/LlvmLinkOptions.kt]: Maybe.
E.g. https://github.com/llvm-mirror/llvm/commit/e7676fec11b02e4b698b5ffc99e1901246a7bf66
Not sure whether it is supported by GNU ld.

**Pull request merged**